routing/state.go

```diff
@@ -0,0 +1,181 @@
+package routing
+
+import (
+	"bytes"
+	"errors"
+	"github.com/gogo/protobuf/proto"
+	"github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/libp2p/go-libp2p-core/peer"
+	pb "github.com/libp2p/go-libp2p-core/routing/pb"
+	ma "github.com/multiformats/go-multiaddr"
+	"time"
+)
+
+// The domain string used for routing state records contained in a SignedEnvelope.
+const StateEnvelopeDomain = "libp2p-routing-state"
+
+// The type hint used to identify routing state records in a SignedEnvelope.
+// TODO: register multicodec
+var StateEnvelopePayloadType = []byte("/libp2p/routing-state-record")
+
+// AnnotatedAddr will extend the Multiaddr type with additional metadata, as
+// extensions are added to the routing state record spec. It's defined now to
+// make refactoring simpler in the future.
+type AnnotatedAddr struct {
+	ma.Multiaddr
+}
```

Comment:
This seems redundant for now; adding fields to the struct is a breaking change and it makes it awkward to work with.
So  if we ever introduce annotations to the multiaddr, we can make the annotated addr struct just in time.